blocks/snippets.py

```diff
@@ -0,0 +1,14 @@
+"""Routines to build frequently used Theano expressions."""
+from theano import tensor
+
+
+def L2_norm(tensors):
+    """Computes the total L2 norm of a set of tensors.
+
+    Parameters
+    ----------
+    tensors : iterable of :class:`~theano.Variable`
+        The tensors.
+
+    """
+    return tensor.sqrt(sum((t ** 2).sum() for t in tensors))
```

Comment:
Oh, but that's the (internal) documentation of the Op for element-wise operations for some reason! The [actual docstring](http://deeplearning.net/software/theano/library/tensor/basic.html#tensor.sqr) is this: "Returns a variable representing the square of a, ie a^2." :)

That reasoning does not make sense to me. Calling a file "snippets" because you don't want to use snippets is like calling a file "old_functions" and filling it with new functions that replace the old ones... Either way, a "snippet" is used to refer to a piece of code that you *don't* factor out, but that you copy and paste all the time. We don't copy and paste and *do* factor out, so they're not snippets, per definition.